src/domains/hoareDomain.ml

```diff
@@ -27,10 +27,10 @@ struct
       | [] -> [e]
       | x::xs -> try op e x :: xs with Lattice.Uncomparable -> x :: join op e xs
 
-    (* widen element e with bucket using op *)
+    (* widen new(!) element e with old(!) bucket using op *)
     let rec widen op e = function
       | [] -> []
-      | x::xs -> try if E.leq e x then [op e x] else widen op e xs with Lattice.Uncomparable -> widen op e xs (* only widen if valid *)
+      | x::xs -> try if E.leq x e then [op x e] else widen op e xs with Lattice.Uncomparable -> widen op e xs (* only widen if valid *)
```

Comment:
The exception should be thrown by the `E.leq`, right? Maybe one can make that more obvious by doing  a match on the result of `leq` and to match both `false` and `exception Lattice.Uncomparable` to `widen op e xs`.

Reply:
`E.leq` (which is `Lval.NormalLval.leq` for the address domain) doesn't throw anything, it just returns `false` for the uncomparable values. It's `op` that throws.
Also they aren't duplicating each other because not raising `Uncomparable` doesn't yet mean that the underlying widening is correctly called (arguments are not switched or it's another unrelated value that happens to be in the same `hash` bucket.